packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/clientcore/ClientCoreWrapperClientMethodTemplate.java

```diff
@@ -26,6 +35,27 @@ public static WrapperClientMethodTemplate getInstance() {
         return INSTANCE;
     }
 
+    @Override
+    public void write(ClientMethod clientMethod, JavaType typeBlock) {
+        ClientMethodType methodType = clientMethod.getType();
+        List<ClientMethodParameter> inputParams = clientMethod.getMethodInputParameters();
+        boolean hasRequestContext
+            = inputParams != null && inputParams.contains(ClientMethodParameter.REQUEST_CONTEXT_PARAMETER);
+
+        if (methodType == ClientMethodType.SimpleSync) {
+            writeConvenienceMethod(clientMethod, typeBlock, ReturnType.SINGLE,
+                clientMethod.getProxyMethod().getSimpleRestResponseMethodName());
+            return;
+        } else if (methodType == ClientMethodType.PagingSync && !hasRequestContext) {
```

Comment:
wonder why do we add this check `!hasRequestContext` and it only applies to paging and lro? Could we add some comments to explain in the code as well?